File: FindAllNumbersDisappearedInAnArray/FindAllNumDis.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
  let arr = [];
  let set = new Set(nums);
  for (let i = 1; i <= nums.length; i++) {
    if (!set.has(i)) {
      arr.push(i);
    }
  }
  return arr;
};

File: FindAllNumbersDisappearedInAnArray/FindAllNumDis.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        length = len(nums)
        num_set = set(range(1, length + 1))
        for i in nums:
            if i in num_set:
                num_set.remove(i)
        return list(num_set)    